### PR TITLE
K9

### PR DESCRIPTION
**Pull request by kinuesantos**

### CODE REVIEW

File: lesson4/backend28.js [hunk 1]
```diff
@@ -5,14 +5,20 @@ app.listen(3618);
 app.use(express.static('public'));
 const bodyParser = require('body-parser');
 app.use(bodyParser.json({limit: '500mb'}));
+const gm = require('gm');
 
 app.post('/send',(req,res)=>{
   const firstData = req.body.name;
```
Comment (chocobrownie) [inline lesson4/backend28.js]: try to change firstData and secondData to more meaningful names

File: lesson4/backend28.js [hunk 2]
```diff
@@ -5,14 +5,20 @@ app.listen(3618);
...
   const secondData = req.body.comment;
   const pic = req.body.newPicData;
-  fs.appendFile('/home/kinue/garageScript/lesson4/public/storing.txt',`${firstData} : ${secondData}\n`)
+  const storingTxt = '/home/kinue/garageScript/lesson4/public/storing.txt';
+  const memePath = '/home/kinue/garageScript/lesson4/public/MemePhotos/' + firstData + '.png';
+  
+  fs.appendFile(storingTxt,`${firstData} : ${secondData}\n`)
```
Comment (chocobrownie) [inline lesson4/backend28.js]: What is this line for? Explain it here or personaly

File: lesson4/backend28.js [hunk 3]
```diff
@@ -5,14 +5,20 @@ app.listen(3618);
...
 
-  fs.writeFile('/home/kinue/garageScript/lesson4/public/selfies.png',`${pic}`, 'base64') 
-  res.send('Pic taken');
+  fs.writeFile(memePath, `${pic}`, 'base64', (err) =>{
```
Comment (chocobrownie) [inline lesson4/backend28.js]: what do you mean by `${pic}`? My guess is you don't understand how expression fully works. Talk to me if you are confused about this

File: lesson4/backend28.js [hunk 4]
```diff
@@ -5,14 +5,20 @@ app.listen(3618);
...
+    gm(memePath).fontSize(40).drawText(50, 50, secondData).write(memePath,() =>  {})
+  console.log('Meme ERROR', err)
```
Comment (chocobrownie) [inline lesson4/backend28.js]: write an if statement here, only console.log error when error exists. Something like if(error){do something}

File: lesson4/backend28.js (hunk 1 above)
Comment (kinuesantos) [reply lesson4/backend28.js]: done

File: lesson4/backend28.js (hunk 2 above)
Comment (kinuesantos) [reply lesson4/backend28.js]: I'll explain personaly

File: lesson4/backend28.js (hunk 3 above)
Comment (kinuesantos) [reply lesson4/backend28.js]: ok, I'll explain this one as well once you get here.

File: lesson4/backend28.js [hunk 5]
```diff
@@ -5,14 +5,20 @@ app.listen(3618);
 app.use(express.static('public'));
 const bodyParser = require('body-parser');
 app.use(bodyParser.json({limit: '500mb'}));
+const gm = require('gm');
 
 app.post('/send',(req,res)=>{
-  const firstData = req.body.name;
-  const secondData = req.body.comment;
+  const names = req.body.name;
```
Comment (saumya665) [inline lesson4/backend28.js:11]: optional comment: don't need to save names and comments as variables.

**Review by saumya665 — APPROVED** [review]

File: lesson4/backend28.js [hunk 6]
```diff
@@ -5,14 +5,20 @@ app.listen(3618);
...
+  const comments = req.body.comment;
   const pic = req.body.newPicData;
-  fs.appendFile('/home/kinue/garageScript/lesson4/public/storing.txt',`${firstData} : ${secondData}\n`)
+  const storingTxt = '/home/kinue/garageScript/lesson4/public/storing.txt';
+  const memePath = '/home/kinue/garageScript/lesson4/public/MemePhotos/' + names + '.png';
```
Comment (saumya665) [inline lesson4/backend28.js]: can directly use req.query.name here

**Review by allopez7 — APPROVED** [review]
Looks good!

**Pull request merged**